### DISCUSSION TITLE
React Strict Mode is disabled after installation in some cases

### DISCUSSION THREAD

**Discussion by angry-meow** [vercel/next.js#45186] [Help]
As you know, `create-next-app` suggests 5 possible options for installation.

* TypeScript usage
* ESLint usage
* `src/` directory usage
* `app/` directory usage
* Aliases usage

I tried every possible options combination (except aliases), installed Next.js 16 times and found out that when you install Next.js with the experimental `app/` directory, the `nextConfig` object doesn't have `reactStrictMode: true` property in it.

Who knows, why?

**Comment by icyJoseph** [comment]
That's 32 combinations right? Wow! 

Looks like when the idea of "templates", either 'app' or 'page', was introduced, the template for 'app' had a `next.config.js` without strict mode, as this PR shows https://github.com/vercel/next.js/pull/41660/files, and you can see in the current template version:

- app TS template: https://github.com/vercel/next.js/blob/canary/packages/create-next-app/templates/app/ts/next.config.js
- app JS template: https://github.com/vercel/next.js/blob/canary/packages/create-next-app/templates/app/js/next.config.js

The PR doesn't have any comments really, so I can't go off anything there... Might be worth to open an Pull Request!

## Edit

It looks like the strict mode is true by default, without having to include it in the `next.config.js` file. Whether that's good or bad, its an opinion.

**Comment by kfirfitousi** [comment]
`reactStrictMode` is true by default when using the `/app` directory.
Lee Robinson mentioned it on this video: https://youtu.be/eFzNekhVhmM?t=795

**Reply by guitheengineer** [reply]
Yeah, relevant code:

https://github.com/vercel/next.js/blob/d356458202162b64eadfd51c3ff4662d624cccbf/packages/next/src/build/webpack-config.ts#L265-L272

**Reply by icyJoseph** [reply]
Oh wow, might be worth adding that to the beta docs, it might not be obvious for some that the Strict Mode is on, as it happened to OP.
